base-server/index.js

```diff
@@ -719,11 +739,23 @@ class BaseServer {
     let processor = this.types[type]
     if (processor) {
       return processor
+    } else if (Object.keys(this.regexTypes).length > 0) {
+      return this.getRegexProcessor(type) || this.otherProcessor
     } else {
       return this.otherProcessor
     }
   }
 
+  getRegexProcessor (type) {
+    let regexps = Object.keys(this.regexTypes).filter(
+      key => key.startsWith('/') && key.endsWith('/') && key.length > 1
+    )
+    let match = regexps.find(
+      regexp => type.match(regexp.substr(1, regexp.length - 2)) !== null
```

Comment:
We can use `this.regexTypes = Map<RegExp, callbacks>()` to reduce these hacks.

```js
for (let regexp of this.regexTypes.keys()) {
  if (type.match(regexp)) {
    return this.regexTypes.get(regexp)
  }
}
```

Reply:
haven't though of that, cool, this is much better